change/@ni-nimble-components-e476eedb-989a-4246-811b-039ca8a8352e.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Opening a comment to get nimble team thoughts on the below issue in the editor:

**Issue**:
1. In nimble-toolbar, button focusing operates based on the tabindex value. However, when I click (mouse down) on another button with tabindex=-1, the button with tabindex=0 receives focus during the mouse down and loses it on the mouse up. In the example below, the bold button has tabindex=0, italics have tabindex=-1, and clicking (mouse down) causes focus to shift to the bold button. It is because after clicking the button, we will refocus on the editor to have a seamless editing experience. On bold button click,
```js 
this.tiptapEditor.chain().focus().toggleBold().run();
```

![image](https://github.com/ni/nimble/assets/123377523/6e4e8f36-055b-4aa4-a604-c365a4472548)

This issue is reproducible only in the Chrome browser and it might be due to the Chromium bug on the delegate focus where the toggle button uses the same as mentioned in the comment of the workaround in [anchor-menu-item file](https://github.com/ni/nimble/blob/158b1d29e9cb8d6591662404ee4ed0921b3e1361/packages/nimble-components/src/anchor-menu-item/index.ts#L70C1-L76C98). We are not sure whether it is related to this issue but would like to hear your thoughts to debug this in a right direction for a faster resolution.

2. On click of the `nimble-toggle-button` with just the icon in it, the tabindex value is not updated to 0 if I click on the icon (marked red in the below image). But if I click outside the icon, the tabindex value gets updated to 0. It is reproducible in the existing nimble-toolbar when only the icon is enabled. Please let us know if it is something to fix in the underlying component or in the consumer component.
![image](https://github.com/ni/nimble/assets/123377523/94be9572-dc99-443d-98cc-f654b4955a75)

Reply:
I can reproduce both of the issues you pointed out. Issue 1 is not related to the Chromium bug affecting the anchor-menu-item, because that is triggered by `tabindex` being set, which it isn't because of issue 2. Issue 1 is caused by the toolbar getting focus when clicking a button that is not focusable (i.e. not the one with `tabindex=0`). I can even reproduce it in our toolbar story (if I add another control):
![toolbar_focus](https://github.com/ni/nimble/assets/7282195/24c12137-3fce-43a8-a96f-3b4af0ad6a3f)
It is interesting that this only happens in Chrome. Please file Nimble a bug about the wrong toolbar button getting focused when the toolbar recieves focus. I suspect the cause is a bug in Chrome, but we should track it.

If we could fix issue 2, that would also resolve/workaround issue 1, because the clicked button would be getting `tabindex=0`, and thus the focus. Please file a separate Nimble bug for this. We should try to find a fix or workaround for you soon.

Reply:
Sure, thanks for the details. I will create separate bugs for the above issues.